pkg/operator/ceph/nvmeof/spec.go

```diff
@@ -0,0 +1,451 @@
+/*
+Copyright 2025 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nvmeof
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	cephclient "github.com/rook/rook/pkg/daemon/ceph/client"
+	cephconfig "github.com/rook/rook/pkg/operator/ceph/config"
+	"github.com/rook/rook/pkg/operator/ceph/controller"
+	"github.com/rook/rook/pkg/operator/k8sutil"
+	apps "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	// AppName is the name of the app
+	AppName             = "rook-ceph-nvmeof"
+	nvmeofIOPort        = 4420
+	nvmeofGatewayPort   = 5500
+	nvmeofMonitorPort   = 5499
+	nvmeofDiscoveryPort = 8009
+	defaultNVMeOFImage  = "quay.io/ceph/nvmeof:1.5"
```

Comment:
I'm curious about @travisn 's thoughts, but I would suggest not having a default here. If we keep a default, we will always be obligated to keep it up to date. If we require it as input, we can avoid a minor, repeated (and probably often overlooked) maintenance task

Reply:
I've added an optional `image` parameter to the NVMeOFGatewaySpec. If not specified, it defaults to `quay.io/ceph/nvmeof:1.5`. This allows users to override the image when needed while maintaining backward compatibility.